src/utilities/fileSystem.utils.ts

```diff
@@ -37,18 +39,41 @@ export const listFilesInDirectory = async (dirPath: string, includeDirectories:
 	return files;
 };
 
-export const readDirectoryRecursive = (directoryPath: string): string[] => {
+const isUnixHiddenPath = function (path: string) {
+	return /(^|\/)\.[^\/\.]/g.test(path);
+};
+
+const isWinHiddenPath = function (path: string) {
+	const pathAttrs = winattr.getSync(path);
+	return pathAttrs.hidden;
+};
+
+export const readDirectoryRecursive = async (directoryPath: string): Promise<string[]> => {
 	let files: string[] = [];
+	const isWin = process.platform === "win32";
 
-	fs.readdirSync(directoryPath).forEach((file) => {
-		const fullPath = path.join(directoryPath, file);
-		if (fs.statSync(fullPath).isDirectory()) {
-			files = files.concat(readDirectoryRecursive(fullPath));
-		} else {
-			files.push(fullPath);
-		}
-	});
+	const readDirSync = (dirPath: string) => {
```

Comment:
technically using map is more suitable, since you are mapping the list 
feel free to change it in the future if you want
you don't have to

Reply:
Created a ticket:
https://linear.app/autokitteh/issue/UI-176/refactor-recursivefiledirectory-function-to-use-map